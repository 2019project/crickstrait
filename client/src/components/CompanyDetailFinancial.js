import React, { Component } from 'react'
import CompanyDetailSecondaryNavbar from './Common/CompanyDetailSecondaryNavbar'
import { getCompanyDetailById } from '../actions/CompanyDetail'
import { connect } from 'react-redux'
import '../styles/CompanyDetailFinancial.css'

export class CompanyDetailFinancial extends Component {
  componentDidMount () {
    // getting the id from the params
    const id = this.props.match.params.id
    // PASSING THE ID TO THE ACTION
    this.props.getCompanyDetailById(id)
  }
  render () {
    return (
      <div>
        {/* CALLING THE SECONDARY NAVBAR  */}
        <CompanyDetailSecondaryNavbar />
        <h1 id='financialsh1'>Balance Sheet </h1>
        <div id='table-container'>
          <table id='table_data'>
            {this.props.company.balancesheet ? (
              <>
                {this.props.company.balancesheet.map(balance => (
                  <>
                    <div id='company-details-date' />
                    <tr>
                      <td id='comp-dates'>{balance['ticker_dates']}</td>

                      <td id='cashequivalents'>
                        {balance['Cash and Cash Equivalents']}
                      </td>
                      <td>{balance['Current Assets']}</td>
                      <td>{balance['Total Assets']}</td>
                      <td>{balance['Accounts Payable']}</td>
                      <td>{balance['Receivables']}</td>
                      <td>{balance['Total Liabilities']}</td>
                      <td>{balance['Current Liabilities']}</td>
                      <td>{balance['Preferred Equity']}</td>
                      <td>{balance['Equity Before Minorities']}</td>
                      <td>{balance['Minorities Interest']}</td>
                      <td>{balance['Noncurrent Liabilities']}</td>
                    </tr>
                  </>
                ))}
              </>
            ) : (
              <p />
            )}
          </table>
        </div>

        <h1 id='financialsh1'>Cash Flow </h1>
        <div id='table-container'>
          <table id='table_data'>
            {this.props.company.cashflow ? (
              <>
                {this.props.company.cashflow.map(cashflow => (
                  <>
                    <tr>
                      <td id='comp-dates'>{cashflow['ticker_dates']}</td>

                      <td id='current-assests'>
                        {cashflow['Cash From Operating Activities']}
                      </td>
                      <td>{cashflow['Cash From Investing Activities']}</td>
                      <td>{cashflow['Cash From Financing Activities']}</td>
                      <td>{cashflow['EBITDA']}</td>
                      <td>{cashflow['Net Change in Cash']}</td>
                      <td>{cashflow['Net PP&E']}</td>
                      <td>{cashflow['Dividends']}</td>
                    </tr>
                  </>
                ))}
              </>
            ) : (
              <p />
            )}
          </table>
        </div>

        <h1 id='financialsh1'>Profit And Loss </h1>
        <div id='table-container'>
          <table id='table_data'>
            {this.props.company.cashflow ? (
              <>
                {this.props.company.profitandloss.map(profitandloss => (
                  <>
                    <tr>
                      <td id='comp-dates'>{profitandloss['ticker_dates']}</td>

                      <td>{profitandloss['Revenues']}</td>
                      <td>{profitandloss['EBIT']}</td>
                      <td>{profitandloss['Net Profit']}</td>
                    </tr>
                  </>
                ))}
              </>
            ) : (
              <p />
            )}
          </table>
        </div>

        <h1 id='financialsh1'>Ratios </h1>
        <table id='table_data'>
          {this.props.company.cashflow ? (
            <>
              {this.props.company.ratios.map(ratios => (
                <>
                  <tr>
                    <td id='comp-dates'>{ratios['ticker_dates']}</td>
                    {/* mapping only the first two values and ignoring the rest of the values  */}
                    <td>{Number(ratios['Current Ratio']).toFixed(2)}</td>
                    <td>
                      {Number(ratios['Liabilities To Equity']).toFixed(2)}
                    </td>
                    <td>{Number(ratios['Debt To Asset']).toFixed(2)}</td>
                  </tr>
                </>
              ))}
            </>
          ) : (
            <p />
          )}
        </table>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  company: state.CompanyDetailReducer.company
})
export default connect(
  mapStateToProps,
  { getCompanyDetailById }
)(CompanyDetailFinancial)
