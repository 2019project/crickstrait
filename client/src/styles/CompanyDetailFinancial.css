th,
td {
  text-align: left;
  padding: 8px;
}
table {
  border-collapse: collapse;
  border-spacing: 0;
  width: 100%;
  /* border: 1px solid red; */
  padding: 10px 10px 10px 10px;
}
.table_data {
  /* onlythe table will scroll */
  overflow-x: scroll;
}
/* tr:nth-child(even) {
  background-color: #f2f2f2;
} */

/* converting row to column  */
table tr > * {
  display: block;
}
table tr {
  display: table-cell;
}

#bla {
  /* font-size: 100px; */
  border: 1px solid red;
  background-color: #2c3e50;

  border-radius: 10px 0px;
}
#balance {
  width: 700px;
}
#company-details-date {
  background-color: #2c3e50;
}
#comp-dates1 {
  background-color: #2c3e50;
  border-radius: 10px;
}
#comp-dates {
  background-color: #2c3e50;
  color: white;
  /* border-radius: 10px; */
}
#table-container {
  padding: 10px 10px 10px 10px;
}
