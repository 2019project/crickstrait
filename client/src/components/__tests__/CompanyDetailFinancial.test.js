import React from "react";
import { shallow } from "enzyme";
import { CompanyDetailFinancial } from "../CompanyDetailFinancial";

const getCompanyDatesById = jest.fn();
const getByYear = jest.fn();

const wrapper = shallow(
  <CompanyDetailFinancial
    getCompanyDatesById={getCompanyDatesById}
    getByYear={getByYear}
    match={{ params: { id: 1 } }}
  />
);

describe("Testing Company Detail Financial Component", () => {
  it("should mount the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should display the market cap", () => {
    expect(wrapper.find("#table1").text()).toBe("Balance Sheet");
  });
  it("should display the Sector", () => {
    expect(wrapper.find("#table2").text()).toBe("Cash Flow");
  });
  it("should display the industry", () => {
    expect(wrapper.find("#table3").text()).toBe("Profit And Loss");
  });
  it("should display the employee", () => {
    expect(wrapper.find("#table4").text()).toBe("Ratios");
  });

  it("should simulate button click on balance Sheet Option button tab ", () => {
    wrapper.find("#item1").simulate("click");
    expect(wrapper.state().balanceSheetOption).toBe(true);
    expect(wrapper.state().cashflowOption).toBe(false);
    expect(wrapper.state().profitLossOption).toBe(false);
    expect(wrapper.state().ratiosOption).toBe(false);
  });
  it("should simulate button click on cash flow Option button tab ", () => {
    wrapper.find("#item2").simulate("click");
    expect(wrapper.state().balanceSheetOption).toBe(false);
    expect(wrapper.state().cashflowOption).toBe(true);
    expect(wrapper.state().profitLossOption).toBe(false);
    expect(wrapper.state().ratiosOption).toBe(false);
  });
  it("should simulate button click on profit Loss Option button tab ", () => {
    wrapper.find("#item3").simulate("click");
    expect(wrapper.state().balanceSheetOption).toBe(false);
    expect(wrapper.state().cashflowOption).toBe(false);
    expect(wrapper.state().profitLossOption).toBe(true);
    expect(wrapper.state().ratiosOption).toBe(false);
  });
  it("should simulate button click on ratios Option button tab ", () => {
    wrapper.find("#item4").simulate("click");
    expect(wrapper.state().balanceSheetOption).toBe(false);
    expect(wrapper.state().cashflowOption).toBe(false);
    expect(wrapper.state().profitLossOption).toBe(false);
    expect(wrapper.state().ratiosOption).toBe(true);
  });

  it("should check for presense of onSearch function", () => {
    let getByYear = jest.spyOn(wrapper.instance(), "getByYear");
    expect(getByYear).toBeTruthy();
  });

  it("checks for on getByYear to be called", () => {
    const e = { preventDefault: () => {} };
    jest.spyOn(e, "preventDefault");
    wrapper.find("#financialsYearsOptionID").simulate("click", e);
  });
  it("should check for All years Data", () => {
    const e = {
      target: {
        name: "yearInput",
        value: "All"
      }
    };
    expect(wrapper.state().search_term).toBe(undefined);
    wrapper.instance().getByYear(e);
    expect(getCompanyDatesById).toBeCalledTimes(2);
  });

  it("should check for a single year data", () => {
    const e = {
      target: {
        name: "yearInput",
        value: "2019"
      }
    };
    expect(wrapper.state().search_term).toBe(undefined);
    wrapper.instance().getByYear(e);
    expect(getCompanyDatesById).toBeCalledTimes(3);
  });
});
