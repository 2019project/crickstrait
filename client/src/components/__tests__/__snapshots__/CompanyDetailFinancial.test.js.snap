// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Company Detail Financial Component should mount the component 1`] = `
<div>
  <Connect(CompanyDetailSecondaryNavbar)
    selected="financial"
  />
  <div
    className="financialYearsDropdown"
  >
    <select
      className="financialsYearsOption"
      id="financialsYearsOptionID"
      name="yearInput"
      onChange={[Function]}
      type="text"
      value="All"
    >
      <option
        name="choice"
      >
        All
      </option>
      <option
        key="year0"
        value={2019}
      >
        2019
      </option>
      <option
        key="year1"
        value={2018}
      >
        2018
      </option>
      <option
        key="year2"
        value={2017}
      >
        2017
      </option>
      <option
        key="year3"
        value={2016}
      >
        2016
      </option>
      <option
        key="year4"
        value={2015}
      >
        2015
      </option>
      <option
        key="year5"
        value={2014}
      >
        2014
      </option>
      <option
        key="year6"
        value={2013}
      >
        2013
      </option>
      <option
        key="year7"
        value={2012}
      >
        2012
      </option>
      <option
        key="year8"
        value={2011}
      >
        2011
      </option>
      <option
        key="year9"
        value={2010}
      >
        2010
      </option>
      <option
        key="year10"
        value={2009}
      >
        2009
      </option>
      <option
        key="year11"
        value={2008}
      >
        2008
      </option>
      <option
        key="year12"
        value={2007}
      >
        2007
      </option>
      <option
        key="year13"
        value={2006}
      >
        2006
      </option>
      <option
        key="year14"
        value={2005}
      >
        2005
      </option>
      <option
        key="year15"
        value={2004}
      >
        2004
      </option>
      <option
        key="year16"
        value={2003}
      >
        2003
      </option>
      <option
        key="year17"
        value={2002}
      >
        2002
      </option>
      <option
        key="year18"
        value={2001}
      >
        2001
      </option>
      <option
        key="year19"
        value={2000}
      >
        2000
      </option>
    </select>
  </div>
  <div
    className="financialsMainGrid"
  >
    <sidebar
      className="financialsSidebar"
    >
      <div
        className="financialsSidebarOptions"
        id="financialsSidebarMain"
      >
        <a
          className="financialsSidebarItem"
          href="#table1"
          id="item1"
          onClick={[Function]}
        >
          <img
            alt="Balance Sheet"
            height="80px"
            src="bl.png"
            style={
              Object {
                "margin": "20px 10px 10px 10px",
              }
            }
            width="80px"
          />
          <p
            className="financialsSidebarOptionsP"
          >
            Balance Sheet
          </p>
        </a>
        <a
          className={null}
          href="#table2"
          id="item2"
          onClick={[Function]}
        >
          <img
            alt="Cash Flow"
            height="80px"
            src="cf.png"
            style={
              Object {
                "margin": "20px 10px 10px 10px",
              }
            }
            width="80px"
          />
          <p
            className="financialsSidebarOptionsP"
          >
            Cash Flow
          </p>
        </a>
        <a
          className={null}
          href="#table3"
          id="item3"
          onClick={[Function]}
        >
          <img
            alt="Profit & Loss"
            height="80px"
            src="pl.png"
            style={
              Object {
                "margin": "20px 10px 10px 10px",
              }
            }
            width="80px"
          />
          <p
            className="financialsSidebarOptionsP"
          >
            Profit & Loss
          </p>
        </a>
        <a
          className={null}
          href="#table4"
          id="item4"
          onClick={[Function]}
        >
          <img
            alt="Ratios"
            height="80px"
            src="ratios.png"
            style={
              Object {
                "margin": "20px 10px 10px 10px",
              }
            }
            width="80px"
          />
          <p
            className="financialsSidebarOptionsP"
          >
            Ratios
          </p>
        </a>
      </div>
    </sidebar>
    <div
      className="financialsAllTablesDiv"
    >
      <h1
        className="financialsTitle"
        id="table1"
      >
        Balance Sheet
      </h1>
      <Reports
        headers={
          Array [
            "date",
            "Current Assets",
            "Intangible Assets",
            "Cash & Cash Equivalents",
            "Goodwill",
            "Total Assets",
            "Accounts Payable",
            "Receivables",
            "Current Liabilities",
            "Total Liabilities",
            "Preferred Equity",
            "Total Noncurrent Liabilities",
            "Equity Before Minorities",
          ]
        }
        reportdata={Array []}
        yearValue="All"
      />
      <h1
        className="financialsTitle"
        id="table2"
      >
        Cash Flow
      </h1>
      <Reports
        headers={
          Array [
            "date",
            "Depreciation & Amortisation",
            "PP&E & Intangibles",
            "Net Change in PP&E & Intangibles",
            "Net Change in Cash",
            "Cash From Operating Activities",
            "Cash From Investing Activities",
            "Cash From Financing Activities",
            "Change in Working Capital",
          ]
        }
        reportdata={Array []}
        yearValue="All"
      />
      <h1
        className="financialsTitle"
        id="table3"
      >
        Profit And Loss
      </h1>
      <Reports
        headers={
          Array [
            "date",
            "Revenues",
            "EBIT",
            "Net Profit",
            "R&D",
            "Income Taxes",
          ]
        }
        reportdata={Array []}
        yearValue="All"
      />
      <h1
        className="financialsTitle"
        id="table4"
      >
        Ratios
      </h1>
      <Reports
        headers={
          Array [
            "date",
            "Liabilities to Equity Ratio",
            "Debt to Assets Ratio",
            "Current Ratio",
            "EV / EBITDA",
            "EV / Sales",
            "Operating Income / EV",
          ]
        }
        reportdata={Array []}
        yearValue="All"
      />
    </div>
  </div>
</div>
`;
